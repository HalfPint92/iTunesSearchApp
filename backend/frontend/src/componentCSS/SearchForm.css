/* Styling for Search Form */
.search-form {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-bottom: 20px;
}

.search-input {
  margin-right: 10px;
  padding: 8px;
  border: 1px solid #ddd;
  border-radius: 4px;
  font-size: 16px;
  margin: 20px;
}

.media-type-select {
  margin-right: 10px;
  padding: 8px;
  border: 1px solid #ddd;
  border-radius: 4px;
  font-size: 16px;
}

.search-button {
  padding: 8px 16px;
  background-color: #007bff;
  color: #fff;
  border: none;
  border-radius: 4px;
  font-size: 16px;
  cursor: pointer;
}

.search-button:hover {
  background-color: #0069d9;
}

/* Optional: Add a box-shadow effect to the form */
.search-form {
  box-shadow: 0 2px 4px #0000001a;
}