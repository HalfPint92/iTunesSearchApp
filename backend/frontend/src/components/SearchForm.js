import React, { useState } from 'react';
import './../componentCSS/SearchForm.css';

const SearchForm = ({ onSearch }) => {
  // State for the search term
  const [searchTerm, setSearchTerm] = useState('');
  // State for the selected media type
  const [mediaType, setMediaType] = useState('all');

  const handleChangeSearchTerm = (e) => {
    // Update the search term state when the input value changes
    setSearchTerm(e.target.value);
  };

  const handleChangeMediaType = (e) => {
    // Update the media type state when the select value changes
    setMediaType(e.target.value);
  };

  const handleSubmit = (e) => {
    // Prevent the default form submission behaviour
    e.preventDefault();
    // Call the onSearch function with the search term and media type as arguments
    onSearch(searchTerm, mediaType);
  };

  return (
    <form className="search-form" onSubmit={handleSubmit}>
      <input
        type="text"
        className="search-input"
        placeholder="Enter search term"
        value={searchTerm}
        onChange={handleChangeSearchTerm}
      />
      {/* Input element for entering the search term */}

      <select className="media-type-select" value={mediaType} onChange={handleChangeMediaType}>
        <option value="all">All</option>
        <option value="movie">Movie</option>
        <option value="podcast">Podcast</option>
        <option value="music">Music</option>
        <option value="audiobook">Audiobook</option>
        <option value="shortFilm">Short Film</option>
        <option value="tvShow">TV Show</option>
        <option value="software">Software</option>
        <option value="ebook">eBook</option>
      </select>
      {/* Select element for choosing the media type */}

      <button type="submit" className="search-button">Search</button>
      {/* Button to trigger the search */}
    </form>
  );
};

export default SearchForm;
