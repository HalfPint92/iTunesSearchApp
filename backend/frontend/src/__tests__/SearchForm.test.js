import React from 'react';
import { render } from '@testing-library/react';
import SearchForm from '../components/SearchForm';

test('renders SearchForm correctly', () => {
  const { container } = render(<SearchForm />);
  expect(container).toMatchSnapshot();
});