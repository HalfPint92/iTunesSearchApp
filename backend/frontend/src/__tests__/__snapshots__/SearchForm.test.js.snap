// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SearchForm correctly 1`] = `
<div>
  <form>
    <input
      placeholder="Search term"
      type="text"
      value=""
    />
    <select>
      <option
        value="all"
      >
        All
      </option>
      <option
        value="movie"
      >
        Movie
      </option>
      <option
        value="podcast"
      >
        Podcast
      </option>
      <option
        value="music"
      >
        Music
      </option>
      <option
        value="audiobook"
      >
        Audiobook
      </option>
      <option
        value="shortFilm"
      >
        Short Film
      </option>
      <option
        value="tvShow"
      >
        TV Show
      </option>
      <option
        value="software"
      >
        Software
      </option>
      <option
        value="ebook"
      >
        Ebook
      </option>
    </select>
    <button
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
